### PR TITLE
fix(core): Fix schema-files cache busting

### PR DESCRIPTION
**Pull request by eirikurn**
## What

- Make codegen/backend-clients rebuild when `clientConfig.{yml,yaml,json} files change.
- Make codegen/backend-clients rebuild when `openapi.{yml,yaml,json}` files generated by dependent targets change.
- Make codegen/frontend-clients rebuild when api.graphql changes.

## Why

So that changes in those files cache-bust later targets.

### CODE REVIEW

File: scripts/check-tags.ts [hunk 1]
```diff
@@ -11,6 +11,7 @@ interface Project {
 
 const checkTags = async (filePath: string) => {
   const projectText = await readFile(filePath, 'utf8')
+  console.log(filePath)
```
Comment (saevarma) [inline scripts/check-tags.ts:14]: console.log?

**Review by saevarma — APPROVED** [review]
🚀

File: nx.json [hunk 2]
```diff
@@ -114,7 +114,7 @@
     },
     "codegen/backend-client": {
       "dependsOn": ["^codegen/backend-client", "^codegen/backend-schema"],
-      "inputs": ["schema-files", "^schema-files"]
+      "inputs": ["schema-files", "^schema-outputs"]
```
Comment (saevarma) [inline nx.json:117]: I was reviewing Deloittes car recycling PR #12879 and there we are creating backend client for GQL api, not rest. Don't we need to add GQL files here as inputs for backend-client to resolve Nx cache issues that could happen?

Comment (eirikurn) [reply nx.json:117]: Good point. Added 👍

**Review by brynjarorng — APPROVED** [review]
LGTM

**Pull request merged**